executor/common_linux.h

```diff
@@ -1734,6 +1734,156 @@ static int do_sandbox_namespace(void)
 }
 #endif
 
+#if SYZ_EXECUTOR || SYZ_SANDBOX_ANDROID_UNTRUSTED_APP
+#include <attr/xattr.h>
+
+const uid_t AID_NET_BT_ADMIN = 3001;
+const uid_t AID_NET_BT = 3002;
+const uid_t AID_INET = 3003;
+const uid_t AID_EVERYBODY = 9997;
+const uid_t AID_APP = 10000;
+const char* SELINUX_CONTEXT_UNTRUSTED_APP = "u:r:untrusted_app:s0:c512,c768";
+const char* SELINUX_LABEL_APP_DATA_FILE = "u:object_r:app_data_file:s0:c512,c768";
+
+const uid_t UNTRUSTED_APP_UID = AID_APP + 999;
+const gid_t UNTRUSTED_APP_GID = AID_APP + 999;
+const gid_t UNTRUSTED_APP_GROUPS[] = {UNTRUSTED_APP_GID, AID_NET_BT_ADMIN, AID_NET_BT, AID_INET, AID_EVERYBODY};
+const size_t UNTRUSTED_APP_NUM_GROUPS = sizeof(UNTRUSTED_APP_GROUPS) / sizeof(UNTRUSTED_APP_GROUPS[0]);
+
+#define SELINUX_CONTEXT_FILE "/proc/thread-self/attr/current"
+#define SELINUX_XATTR_NAME "security.selinux"
+
+// see: man 3 getcon
+int getcon(char** context)
+{
+	char buffer[512];
+
+	if (context == NULL)
+		return -1;
+
+	int fd = open(SELINUX_CONTEXT_FILE, O_RDONLY);
+
+	if (fd < 0)
+		return -1;
+
+	ssize_t nread = read(fd, buffer, sizeof(buffer));
+
+	if (nread < 0)
+		return -1;
+
+	char* allocation = (char*)malloc(nread + 1);
+	if (allocation == NULL)
+		return -1;
+
+	memcpy(allocation, buffer, nread);
+	allocation[nread] = '\0';
+
+	*context = allocation;
+	return 0;
+}
+
+// see: man 3 freecon
+void freecon(char* context)
+{
+	free(context);
+}
+
+// see: man 3 setcon
+int setcon(const char* context)
+{
+	// Attempt to write the new context
+	int fd = open(SELINUX_CONTEXT_FILE, O_WRONLY);
+
+	if (fd < 0)
+		return -1;
+
+	ssize_t bytes = write(fd, context, strlen(context));
+
+	if (bytes < 0 || (size_t)bytes != strlen(context))
```

Comment:
What is the preferred way, within syzkaller, to compare `ssize_t` to `size_t`?  The current compiler settings throw an error on comparison, and straight casting without checking for `<0` loses data.

Reply:
What you have now looks fine to me. -1 will convert to size_t 0xffffffffffffffff. strlen can't return 0xffffffffffffffff because that would mean that with the 0 terminator it overflows 64-bit address space.
Alternatively you can convert strlen to ssize_t, that won't lose data.

Reply:
Done